File: referencia/api-springboot/src/test/java/dev/ergrato/museo/PiecesServiceTest.java
package dev.ergrato.museo;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class PiecesServiceTest {

    @Test
    @DisplayName("should return trimmed piece when request is valid")
    void shouldReturnTrimmedPieceWhenRequestIsValid() {
        // Arrange
        var request = new PieceRequest("  Guernica ", "Picasso", 1937);

        // Act
        Piece piece = PiecesService.validate(request, 2026);

        // Assert
        assertThat(piece.getName()).isEqualTo("Guernica");
        assertThat(piece.getYear()).isEqualTo(1937);
    }

    @Test
    @DisplayName("should throw ValidationException when year is in the future")
    void shouldThrowWhenYearIsInTheFuture() {
        var request = new PieceRequest("Future", "Nobody", 2027);

        assertThatThrownBy(() -> PiecesService.validate(request, 2026))
                .isInstanceOf(PiecesService.ValidationException.class)
                .hasMessageContaining("future");
    }
}

File: referencia/api-express/tests/pieces-service.test.js
import { describe, expect, it } from 'vitest';
import { validatePiece, ValidationError } from '../src/pieces-service.js';

describe('validatePiece', () => {
  it('should return trimmed data when piece is valid', () => {
    // Arrange
    const data = { name: '  Guernica ', artist: 'Picasso', year: 1937 };

    // Act
    const piece = validatePiece(data, 2026);

    // Assert
    expect(piece).toEqual({ name: 'Guernica', artist: 'Picasso', year: 1937 });
  });

  it('should throw ValidationError when year is in the future', () => {
    const data = { name: 'Future', artist: 'Nobody', year: 2027 };

    expect(() => validatePiece(data, 2026)).toThrow(ValidationError);
  });
});
